src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Figure['shape'] = 'triangle';
 
-}
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.isInvalid()) {
+      throw new Error('Impossible to create triangle with such parameters');
+    }
+  }
+
+  isInvalid(): boolean {
+    const sides = [this.a, this.b, this.c].sort((a, b) => b - a);
+    const isSideTooSmall = sides.some((side) => side <= 0);
+    const isSideTooLarge = sides[0] >= sides[1] + sides[2];
 
-export class Circle {
+    return isSideTooSmall || isSideTooLarge;
+  }
 
+  getArea(): number {
+    const s = [this.a, this.b, this.c].reduce((acc, curr) => acc + curr, 0) / 2;
```

Comment:
what's s?

Reply:
@naziamoff  semiperimeter, and yes I wrote this as 's' because of the line length issues